backend/src/routes/api/integrations/nim/index.ts

```diff
@@ -0,0 +1,84 @@
+import { FastifyReply, FastifyRequest } from 'fastify';
+import { secureAdminRoute } from '../../../../utils/route-security';
+import { KubeFastifyInstance } from '../../../../types';
+import { isString } from 'lodash';
+import { isAppEnabled, getNIMAccount, createNIMAccount, createNIMSecret } from './nimUtils';
+
+module.exports = async (fastify: KubeFastifyInstance) => {
+  const { namespace } = fastify.kube;
+  const PAGE_NOT_FOUND_MESSAGE = '404 page not found';
+
+  fastify.get(
+    '/',
+    secureAdminRoute(fastify)(async (request: FastifyRequest, reply: FastifyReply) => {
+      await getNIMAccount(fastify, namespace)
+        .then((response) => {
+          if (isAppEnabled(response)) {
+            reply.send({ isAppEnabled: true, canEnable: false, error: '' });
+          } else {
+            reply.send({ isAppEnabled: false, canEnable: true, error: '' });
+          }
+        })
+        .catch((e) => {
+          if (e.response?.statusCode === 404) {
+            // 404 error means the Account CRD does not exist, so cannot create CR based on it.
+            if (
+              isString(e.response.body) &&
+              e.response.body.trim() === PAGE_NOT_FOUND_MESSAGE.trim()
+            ) {
+              fastify.log.error(`NIM not installed, ${e.response?.body}`);
+              reply
+                .status(404)
+                .send({ isAppEnabled: false, canEnable: false, error: 'NIM not installed' });
+            } else {
+              fastify.log.error(`NIM account does not exist, ${e.response.body.message}`);
+              reply.send({ isAppEnabled: false, canEnable: true, error: '' });
```

Comment:
We have a problem here... it's the same scenario as line 19

If we don't have an Account, but we have a CRD -- we are here...
If we have an account, but it's not enabled -- we are here...
```
            reply.send({ isAppEnabled: false, canEnable: true, error: '' });
```